benchmark/infrastructure/BenchmarkMain.cpp

```diff
@@ -104,16 +104,20 @@ int main(int argc, char** argv) {
     printUsageAndExit();
   }
 
-  // Did we get any configuration?
-  BenchmarkConfiguration config{};
-
-  if (vm.count("configuration-json")) {
-    // TODO Get rid of the `dump()`, once the branch for general configuration
-    // options was merged.
-    config.setJsonString(fileToJson(jsonConfigurationFileName).dump());
-  }
-  if (vm.count("configuration-shorthand")) {
-    config.addShortHand(shortHandConfigurationString);
+  // Set all the configuration options, if there was any runtime configuration
+  // given.
+  if (vm.count("configuration-json") || vm.count("configuration-shorthand")) {
```

Comment:
Is that semantically correct?
Shouldn't we ALWAYS call `parseConfig`, even when no values were specified?
That way we don't miss any values that Have to be specified at runtime.
So I would suggest dropping the outer `if` in line 109.

Reply:
Good catch! I've completely forgot about that. (Maybe it's an artifact? I'm not 100% sure, if early versions of `ConfigManager` had a problem with allways being called....)

However, wouldn't such a bug fix need it's own branch for organizational purpose? I mean, it got nothing to do with this one and adding it, just because it would be faster, doesn't sound like good practice.

Reply:
Yes, can you please set up this very tiny fix in another PR.